test/jdk/java/awt/print/PrinterJob/ValidatePage/ValidatePage.java

```diff
@@ -21,407 +21,293 @@
+    Label myOrientationLabel;
+
+    protected void displayPageFormatAttributes() {
+        myWidthLabel.setText("Format Width = " + drnd(myPageFormat.getWidth()));
+        myHeightLabel.setText("Format Height = " + drnd(myPageFormat.getHeight()));
+        myImageableXLabel.setText("Format Left Margin = " + drnd(myPageFormat.getImageableX()));
+        myImageableRightLabel.setText("Format Right Margin = " + drnd(myPageFormat.getWidth()
+                - (myPageFormat.getImageableX() + myPageFormat.getImageableWidth())));
+        myImageableWidthLabel.setText("Format ImageableWidth = "
+                + drnd(myPageFormat.getImageableWidth()));
+        myImageableYLabel.setText("Format Top Margin = "
+                + drnd(myPageFormat.getImageableY()));
+        myImageableBottomLabel.setText("Format Bottom Margin = " + drnd(myPageFormat.getHeight()
+                - (myPageFormat.getImageableY() + myPageFormat.getImageableHeight())));
+        myImageableHeightLabel.setText("Format ImageableHeight = "
+                + drnd(myPageFormat.getImageableHeight()));
+        int o = myPageFormat.getOrientation();
+        myOrientationLabel.setText
+                ("Format Orientation = " +
+                        (switch (o) {
+                            case PageFormat.PORTRAIT -> "PORTRAIT";
+                            case PageFormat.LANDSCAPE -> "LANDSCAPE";
+                            case PageFormat.REVERSE_LANDSCAPE -> "REVERSE_LANDSCAPE";
+                            default -> "<invalid>";
+                        }));
+        Paper p = myPageFormat.getPaper();
+        pw.setText("Paper Width = " + drnd(p.getWidth()));
+        ph.setText("Paper Height = " + drnd(p.getHeight()));
+        pglm.setText("Paper Left Margin = " + drnd(p.getImageableX()));
+        pgiw.setText("Paper Imageable Width = " + drnd(p.getImageableWidth()));
+        pgih.setText("Paper Imageable Height = " + drnd(p.getImageableHeight()));
+
+        pgrm.setText("Paper Right Margin = " + drnd(p.getWidth()
+                - (p.getImageableX() + p.getImageableWidth())));
+        pgtm.setText("Paper Top Margin = " + drnd(p.getImageableY()));
+        pgbm.setText("Paper Bottom Margin = " + drnd(p.getHeight()
+                - (p.getImageableY() + p.getImageableHeight())));
     }
 
-   public static void createDialog( )
-    {
-      dialog = new TestDialog( new Frame(), "Instructions" );
-      String[] defInstr = { "Instructions will appear here. ", "" } ;
-      dialog.printInstructions( defInstr );
-      dialog.show();
-      println( "Any messages for the tester will display here." );
+    static String drnd(double d) {
+        d = d * 10.0 + 0.5;
+        d = Math.floor(d) / 10.0;
+        String ds = Double.toString(d);
+        int decimal_pos = ds.indexOf(".");
+        int len = ds.length();
+        if (len > decimal_pos + 2) {
+            return ds.substring(0, decimal_pos + 2);
+        } else {
+            return ds;
+        }
```

Comment:
```suggestion
        return String.format("%.2f", d);
```
I believe it conveys the meaning better.